### PR TITLE
Display and update fields from fromManyObjects relations in Show card

### PR DESCRIPTION
**Pull request by ijreilly**
In this PR, we implement the display and update of fields from fromManyObjects (e.g update Employees for a Company).

Product requirement
- update should be triggered at each box check/uncheck, not at lose of focus

Left to do in upcoming PRs
- add the column in the table views (e.g. column "Employees" on "Companies" table view)
- add "Add new" possibility when there is no records (as is currently exists for "one" side of relations:)
Capture d’écran 2024-06-10 à 17 38 02

- update cache after an update affecting other records (e.g "Listings" have one "Person"; if listing A belonged to Person A but then we attribute listing A to Person B, Person A is no longer owner of Listing A. For the moment that would not be reflected immediatly leading, to potential false information if information is accessed from cache)
- try to get rid of the glitch - we also have it on the task page example. (probably) due to the fact that we are using a recoil state to read, update then re-read 

https://github.com/twentyhq/twenty/assets/51697796/54f71674-237a-4946-866e-b8d96353c458

### CODE REVIEW

**Comment by ijreilly** [comment]
Im wondering if this should be behind a feature flag to widely test it and release it once it is available in the views too

File: packages/twenty-front/src/modules/activities/inline-cell/components/ActivityTargetInlineCellEditMode.tsx [hunk 1]
```diff
@@ -171,15 +171,10 @@ export const ActivityTargetInlineCellEditMode = ({
...
 
-  const handleCancel = () => {
-    closeEditableField();
-  };
-
   return (
     <StyledSelectContainer>
       <MultipleObjectRecordSelect
         selectedObjectRecordIds={selectedTargetObjectIds}
-        onCancel={handleCancel}
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/activities/inline-cell/components/ActivityTargetInlineCellEditMode.tsx:182]: was not used

File: packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts [hunk 2]
```diff
@@ -137,15 +146,20 @@ export const usePersistField = () => {
...
 
           if (fieldIsRelation) {
-            updateRecord?.({
-              variables: {
-                where: { id: entityId },
-                updateOneRecordInput: {
-                  [fieldName]: valueToPersist,
-                  [`${fieldName}Id`]: valueToPersist?.id ?? null,
+            if (fieldIsRelationFromManyObjects) {
+              throw new Error('Cannot update this relation.');
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts:150]: we are not using usePersistField for FromManyObjects as it requires a specific logic that did not feel right to add here

File: packages/twenty-front/src/modules/object-record/relation-picker/components/MultiRecordSelect.tsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+import { SelectableItem } from '@/ui/layout/selectable-list/components/SelectableItem';
+import { SelectableList } from '@/ui/layout/selectable-list/components/SelectableList';
+import { MenuItem } from '@/ui/navigation/menu-item/components/MenuItem';
+import { isDefined } from '~/utils/isDefined';
+
+export const StyledSelectableItem = styled(SelectableItem)`
+  height: 100%;
+  width: 100%;
+`;
+export const MultiRecordSelect = ({
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/object-record/relation-picker/components/MultiRecordSelect.tsx:24]: this logic was mostly extracted from MultipleObjectRecordSelect to be shared with RelationManyFieldInput

File: packages/twenty-front/src/modules/object-record/relation-picker/hooks/useRelationPickerEntitiesOptions.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { useRecoilValue } from 'recoil';
+
+import { useRelationPickerScopedStates } from '@/object-record/relation-picker/hooks/internal/useRelationPickerScopedStates';
+import { useFilteredSearchEntityQuery } from '@/search/hooks/useFilteredSearchEntityQuery';
+
+export const useRelationPickerEntitiesOptions = ({
```
Comment (ijreilly) [inline packages/twenty-front/src/modules/object-record/relation-picker/hooks/useRelationPickerEntitiesOptions.ts:6]: this logic was mostly extracted from SingleEntitySelectMenuItemsWithSearch to be put in common with RelationManyFieldInput

File: packages/twenty-front/src/modules/object-record/record-field/components/FieldInput.tsx [hunk 5]
```diff
@@ -71,7 +73,13 @@ export const FieldInput = ({
       recordFieldInputScopeId={getScopeIdFromComponentId(recordFieldInputdId)}
     >
       {isFieldRelation(fieldDefinition) ? (
-        <RelationFieldInput onSubmit={onSubmit} onCancel={onCancel} />
+        isFieldRelationFromManyObjects(fieldDefinition) ? (
+          <RelationManyFieldInput
+            relationPickerScopeId={`relation-picker-${fieldDefinition.metadata.fieldName}`}
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/object-record/record-field/components/FieldInput.tsx:78]: shouldn't we have an unique id, maybe based on the field id? Also I believe you can use the function `getScopeIdFromComponentId` to build the right shape for a scope id

File: packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts (hunk 2 above)
Comment (thomtrp) [reply packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts:150]: It feels weird to throw here. You have a boolean `isValuePersistable` above that is true if `fieldIsRelation`. Can you simply change for `fieldIsRelation && ! fieldIsRelationFromManyObjects`?

**Review by thomtrp — COMMENTED** [review]
Not the most familiar with that code

**Review by lucasbordeau — COMMENTED** [review]
LGTM, will review it more in depth later

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**